docs/cow-protocol/reference/contracts/core/README.mdx

```diff
@@ -161,3 +161,31 @@ Partner fees are encoded in the app data struct and are accounted for once the o
 From the perspective of the API, two orders with the same parameters and the same owner but different app data are two different valid orders.
 It can happen that the order that is part of the final settlement is the one controlled by the adversary, especially if other order parameters can be changed to create the appearence of an inflated surplus.
 In this case, the order surplus decreases as partner fees are taken from the surplus.
+
+### Loss of funds if a compromised Balancer contract is deployed
+
+A malicious solver can steal the entire amount of a user's trade if the Balancer contract is compromised and the solver processes a vulnerable user order.
+
+Due to a vulnerability in the [profanity vanity address generator](https://medium.com/@rebryk/how-to-hack-a-vanity-address-generated-with-profanity-ffad61ecacd2), the deployer wallet for the Balancer Vault was compromised. This allows arbitrary actors to deploy any bytecode to the address that is normally used to deploy [Balancer's Core Vault contract](https://docs.balancer.fi/concepts/vault/).
+
+In order to preserve a consistent settlement address on all chains, CoW Protocol uses the `CREATE2` opcode to deploy the settlement contract. Since the balancer vault address is supplied as an immutable constructor property, the Settlement contract contains a dependency upon this compromised contract.
+
+The Balancer vault is only used in `GPv2Transfer.sol` as an alternative way for a user to receive or supply funds from the settlement contract when the user sets `buyTokenBalance` or `sellTokenBalance` to `BALANCE_INTERNAL`.
+
+There are two general effects that this vulnerability can have:
+* If the compromised contract at the `Balancer` address reverts upon a call to `manageUserBalance`, the attacker can effectively DoS any trade batches that contain a trade using `BALANCE_INTERNAL`.
```

Comment:
ubernit: I personally would use "prevent users from trading" instead of using "DoS"